core/edit/edit_command_actions.py

```diff
@@ -88,26 +88,37 @@ def edit_action(m) -> EditAction:
 }
 
 
-def run_action_callback(action: EditAction):
-    action_type = action.type
-
-    if action_type in simple_action_callbacks:
-        callback = simple_action_callbacks[action_type]
-        callback()
-        return
-
-    match action_type:
-        case "insert":
-            assert isinstance(action, EditInsertAction)
-            actions.insert(action.text)
-
-        case "wrapWithDelimiterPair":
-            assert isinstance(action, EditWrapAction)
-            return lambda: actions.user.delimiter_pair_wrap_selection(action.pair)
-
-        case "applyFormatter":
-            assert isinstance(action, EditFormatAction)
-            actions.user.formatters_reformat_selection(action.formatters)
-
-        case _:
-            raise ValueError(f"Unknown edit action: {action_type}")
+@mod.action_class
+class Actions:
+    def run_action_callback(action: EditAction):
+        """
+        Run a callback that applies an edit action to the selected
+        Intended for internal use and overwriting
+        """
+        action_type = action.type
+
+        if action_type in simple_action_callbacks:
```

Comment:
This if statements means that you can't add more callbacks to a context implementation of `get_simple_action_callback()`.

Reply:
That's a good catch, I have altered it. I don't think there is any way to call the original mod action function once you have created a context override, correct? So while your modification would let someone entirely replace the callbacks, it would not let them extend it. Any ideas for how to do this? It will actually have an immediate benefit, because I will be able to map ctrl+w to clear word left, which will be faster and cleaner than the current implementation of having to press escape